From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 28 Oct 2017 20:46:43 +0100
Subject: mac80211: Avoid ABI change in 4.9.53
Forwarded: not-needed

Commit 531682159092 ("mac80211: fix VLAN handling with TXQs") replaced
a union with two separate members in ieee80211_tx_info::control.  The
one which changed offset (enqueue_time) doesn't appear to be accessed
outside of the mac80211 module, so this shouldn't be an ABI change for
drivers.  Therefore hide the ABI change from genksyms.

---
--- a/include/net/mac80211.h
+++ b/include/net/mac80211.h
@@ -902,10 +902,19 @@ struct ieee80211_tx_info {
 				unsigned long jiffies;
 			};
 			/* NB: vif can be NULL for injected frames */
+#ifndef __GENKSYMS__
 			struct ieee80211_vif *vif;
+#else
+			union {
+				struct ieee80211_vif *vif;
+				codel_time_t enqueue_time;
+			};
+#endif
 			struct ieee80211_key_conf *hw_key;
 			u32 flags;
+#ifndef __GENKSYMS__
 			codel_time_t enqueue_time;
+#endif
 		} control;
 		struct {
 			u64 cookie;
